plugins/MauticCitrixBundle/EventListener/LeadSubscriber.php

```diff
@@ -0,0 +1,378 @@
+                                                                    $alias.'sub1.event_type',
+                                                                    "'".CitrixEventTypes::REGISTERED."'"
+                                                                ),
+                                                                $q->expr()->in($alias.'sub1.event_name', $eventNames),
+                                                                $q->expr()->eq($alias.'sub1.email', $alias.'.email')
+                                                            )
+                                                        )->getSQL();
+
+                                $queryNbParticipated = $em->getConnection()->createQueryBuilder()
+                                                          ->select('count(*)')
+                                                          ->from($citrixEventsTable, $alias.'sub2')
+                                                          ->where(
+                                                              $q->expr()->andX(
+                                                                  $q->expr()->eq(
+                                                                      $alias.'sub2.event_type',
+                                                                      "'".CitrixEventTypes::ATTENDED."'"
+                                                                  ),
+                                                                  $q->expr()->in($alias.'sub2.event_name', $eventNames),
+                                                                  $q->expr()->eq($alias.'sub2.email', $alias.'.email')
+                                                              )
+                                                          )->getSQL();
+
+                                $subQuery = '(('.$queryNbRegistered.') > ('.$queryNbParticipated.')) AND '.$alias.'.email = l.email';
+
+                                if ('' !== $leadEmail) {
+                                    $subQuery .= ' AND '.$alias.".email='".$leadEmail."'";
+                                }
+
+                                $subQuery = 'EXISTS ( SELECT null FROM '.$citrixEventsTable.' AS '.$alias.' WHERE ( '.$subQuery.'))';
+                            }
+                        }
+                    }
+                } else {
+                    if ($func === 'notIn') {
+                        if ($currentFilter === $product.'-registration') {
+                            $subQuery = 'NOT EXISTS ('.$subQueriesSQL[CitrixEventTypes::REGISTERED].')';
+                        } else {
+                            if ($currentFilter === $product.'-attendance') {
+                                $subQuery = 'NOT EXISTS ('.$subQueriesSQL[CitrixEventTypes::ATTENDED].')';
+                            }
+                        }
+                    }
+                }
+
+                $event->setSubQuery($subQuery);
```

Comment:
I have 664 events in my mautic_plugin_citrix_events table, and this query took 48 seconds to execute for "did not attend" filter: 

```
SELECT count(l.id) as lead_count, max(l.id) as max_id FROM mautic_leads l LEFT JOIN mautic_lead_lists_leads ll ON (ll.leadlist_id = 64) AND (ll.lead_id = l.id) AND (ll.date_added <= '2016-11-24 02:36:34') WHERE (EXISTS ( SELECT null FROM mautic_plugin_citrix_events AS nwXpreiC WHERE ( ((SELECT count(*) FROM mautic_plugin_citrix_events nwXpreiCsub1 WHERE (nwXpreiCsub1.event_type = 'registered') AND (nwXpreiCsub1.event_name IN ('alan-s-test-meeting_#497769037')) AND (nwXpreiCsub1.email = nwXpreiC.email)) > (SELECT count(*) FROM mautic_plugin_citrix_events nwXpreiCsub2 WHERE (nwXpreiCsub2.event_type = 'attended') AND (nwXpreiCsub2.event_name IN ('alan-s-test-meeting_#497769037')) AND (nwXpreiCsub2.email = nwXpreiC.email))) AND nwXpreiC.email = l.email))) AND (ll.lead_id IS NULL)
```

This has to be optimized.